modules/userdirectory/src/main/java/org/opencastproject/userdirectory/UserAndRoleDirectoryServiceImpl.java

```diff
@@ -339,8 +340,17 @@ private User loadUser(Tuple<String, String> orgUser) {
     // Consult roleProviders if this is not an internal system user
     if (!InMemoryUserAndRoleProvider.PROVIDER_NAME.equals(user.getProvider())) {
       for (RoleProvider roleProvider : roleProviders) {
+        String providerOrgId = roleProvider.getOrganization();
+
+        if (!ALL_ORGANIZATIONS.equals(providerOrgId) && !orgUser.getA().equals(providerOrgId)) {
+          continue;
+        }
+
         for (Role role : roleProvider.getRolesForUser(user.getUsername())) {
-          roles.add(JaxbRole.fromRole(role));
+          if (Objects.equals(JaxbOrganization.fromOrganization(user.getOrganization()).getId(),
+                  role.getOrganizationId())) {
+            roles.add(JaxbRole.fromRole(role));
+          }
```

Comment:
The second check should be optional.  Is there a reason why you left it?

Reply:
No particular reason, so I removed it now!